client/me/purchases/confirm-cancel-purchase/index.jsx

```diff
@@ -24,43 +24,36 @@ const ConfirmCancelPurchase = React.createClass( {
 		selectedSite: React.PropTypes.object
 	},
 
-	componentWillMount() {
-		recordPageView( 'confirm_cancel_purchase', this.props );
-	},
-
-	componentWillReceiveProps( nextProps ) {
-		recordPageView( 'confirm_cancel_purchase', this.props, nextProps );
+	getInitialState() {
+		return {
+			isFormLoaded: false
+		};
 	},
 
-	componentDidMount() {
-		if ( ! this.props.selectedPurchase.hasLoadedFromServer || ! this.props.selectedSite ) {
-			return;
-		}
-
-		this.resetEndpointForm();
-	},
-
-	componentDidUpdate( prevProps ) {
-		if ( ! this.props.selectedPurchase.hasLoadedFromServer ) {
-			return;
-		}
-
-		if ( isEqual( prevProps.selectedPurchase.data, this.props.selectedPurchase.data ) &&
-				( prevProps.selectedSite.ID === this.props.selectedSite.ID ) ) {
-			return;
-		}
-
-		this.resetEndpointForm();
+	componentWillMount() {
+		this.loadEndpointForm();
+		recordPageView( 'confirm_cancel_purchase', this.props );
 	},
 
-	resetEndpointForm() {
-		const { selectedSite, selectedPurchase } = this.props;
-
-		loadEndpointForm( {
-			container: React.findDOMNode( this.refs.root ),
-			selectedSite,
-			selectedPurchase: selectedPurchase.data,
-			onSubmit: this.handleSubmit
+	loadEndpointForm() {
+		const purchase = getPurchase( this.props );
+
+		loadEndpointForm( purchase, ( html, initializeForm ) => {
+			if ( ! this.isMounted() ) {
+				return;
+			}
+
+			this.setState( { isFormLoaded: true }, () => {
+				const container = React.findDOMNode( this.refs.root );
+
+				container.innerHTML = html;
```

Comment:
Is there a reason to do it this way rather than with `_dangerouslySetInnerHTML`? The thing I _like_ about `_dangerouslySetInnerHTML` is that it makes it clear that this is a hack, and we'll be nagged to remove it. This way sidesteps the obviously bad `_dangerouslySetInnerHTML` and I worry that it will be more likely to be overlooked.